gradle-plugin/src/main/java/com/microsoft/gradle/GradleProjectModelBuilder.java

```diff
@@ -177,45 +174,55 @@ private DefaultGradleDependencyNode resolveDependency(ResolvedDependencyResult r
 	}
 
 	private List<String> getPlugins(Project project) {
-		Convention convention = project.getConvention();
-		return new ArrayList<>(convention.getPlugins().keySet());
+		// The Convention API was removed in Gradle 8.0+
+		// Use the PluginContainer instead to get applied plugins
+		List<String> plugins = new ArrayList<>();
+		project.getPlugins().forEach(plugin -> {
```

Comment:
Which version is this new api introduced?

Reply:
Yes. Here is an example: https://github.com/microsoft/build-server-for-gradle/blob/87ae5dcdc8b1a5ee99218b409efaec2c382aebcd/plugin/src/main/java/com/microsoft/java/bs/gradle/plugin/SourceSetsModelBuilder.java#L150-L183

Basically, check the current version of gradle, and use reflection to call removed apis